site/app/models/gradeable/LateDayInfo.php

```diff
@@ -161,43 +236,64 @@ public function getStatusMessage() {
         }
     }
 
+    /**
+     * Gets the status messages for this gradeable
+     * @return array
+     */
+    public static function getSimpleMessageFromSatus() {
+        return [
+            self::STATUS_NO_ACTIVE_VERSION => 'No Submission',
+            self::STATUS_GOOD => 'Good',
+            self::STATUS_LATE => 'Late',
+            self::STATUS_BAD => 'Bad'
+        ];
+    }
+
     /**
      * Gets if this user has late days info available (if they have an active version)
      * @return bool
      */
     public function hasLateDaysInfo() {
-        return $this->graded_gradeable->getAutoGradedGradeable()->hasActiveVersion();
+        return $this->has_late_day_info;
     }
 
     /**
-     * Gets the number of late days charged for this assignment
+     * Gets the number of late days charged from this event
      * @return int
      */
     public function getLateDaysCharged() {
-        if ($this->getStatus() === self::STATUS_BAD) {
-            // Don't charge late days for BAD status
-            return 0;
-        }
-        // clamp the days charged to be the days late minus exceptions above zero.
-        return max(0, min($this->getDaysLate(), $this->getLateDaysAllowed()) - $this->getLateDayException());
+        return -$this->late_days_change;
+    }
+
+    /**
+     * Gets the number of late days increased from this event
+     * @return int
+     */
+    public function getLateDaysChange() {
+        return $this->late_days_change;
     }
```

Comment:
If I was reading through the code and encountered `getLateDaysCharged()` I would not immediately expect this function to return a negative number.  Rather, I would expect it to return a positive number representing the number of late days charged.  Similarly, it seems rather confusing to have two functions which return the exact same thing.  Is there a reason for this or should `getLateDaysCharged()` be eliminated altogether?

Just to be clear, this isn't a requested change--just food for thought in case changes are desired.

Reply:
Very good point. I opted to max out the value at 0. I think it could provide better functionality for future changes. I think only late gradeables should return a significant value.